### DISCUSSION TITLE
[Feature] Tab complete based on caret position

### DISCUSSION THREAD

**Discussion by ScrubN** [night/betterttv#6725] [Ideas]
As it currently stands, tab completion only allows completing the last word in the textbox:
https://github.com/night/betterttv/blob/846b3442889cc76ed1736cd290c238df2430d7a7/src/modules/chat_tab_completion/index.js#L74-L78

If it is possible to get the current position of the caret in the textbox, then it would be nice if tab completion worked on words at other positions in the chatbox too.

**Comment by night** [comment]
it isn't easily possible any longer, since resolving caret position and positioning injected strings on Twitch's slate editor is not trivial. this is because they use slate editor, which maintains an internal state that is not the easiest to work around outside of its library.

**Reply by ScrubN** [reply]
I see, that's unfortunate.

Thank you for the quick response.
